crates/api/src/func.rs

```diff
@@ -167,14 +167,24 @@ macro_rules! getters {
             ensure!(params.next().is_none(), "Type mismatch: too many arguments (expected {})", n);
 
             // ... then do the same for the results...
-            let mut results = self.ty().results().iter().cloned();
+            let mut results = ty.results().iter().cloned();
             R::matches(&mut results)
                 .context("Type mismatch in return type")?;
             ensure!(results.next().is_none(), "Type mismatch: too many return values (expected 1)");
 
+            // Pass the instance into the closure so that we keep it live for the lifetime
+            // of the closure. Pass the export in so that we can call it.
+            struct ClosureState {
```

Comment:
You can skip the `struct` type definition here, inside of the closure you can do `drop(&instance_clone)` and just use `export` directly, and that'll capture everything necessary.

Reply:
Ah, cool. I've now done this.